src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -213,13 +289,12 @@ end
 function final_retval(X, AX, resid_history, niter, n_matvec)
     λ = real(diag(X' * AX))
     residuals = AX .- X*Diagonal(λ)
-    (λ=λ, X=X,
+    (λ=Array(λ), X=X,
      residual_norms=[norm(residuals[:, i]) for i in 1:size(residuals, 2)],
-     residual_history=resid_history[:, 1:niter+1],
-     n_matvec=n_matvec)
+     residual_history=resid_history[:, 1:niter+1], n_matvec=n_matvec)
+    #λ doesn't have to be on the GPU, but X does (ψ should always be on GPU throughout the code).
```

Comment:
No reference to DFTK from inside LOBPCG (see https://github.com/JuliaMolSim/DFTK.jl/issues/690)

Reply:
LOBPCG should just return lambda on the GPU if everything is provided there. If you want to get it to an array, do it outside of LOBPCG